src/System.IO.Pipes/src/System/IO/Pipes/PipeStream.Unix.cs

```diff
@@ -58,6 +59,15 @@ internal static string GetPipePath(string serverName, string pipeName)
             // naming scheme used as that breaks the ability for code running on an older
             // runtime to connect to code running on the newer runtime.  That means we're stuck
             // with a tmp file for the lifetime of the server stream.
+            if (isCurrentUserOnly)
+            {
+                string directory = Path.Combine(Path.GetTempPath(), ".NET" + Interop.Sys.GetEUid());
+                Directory.CreateDirectory(directory);
```

Comment:
The directory should be created already with the proper permission settings, not applying them after. Use mkdir directly if needed.

Reply:
Also what happens when the same user creates multiple server pipes? This is perhaps something that should be part of the tests.

Reply:
On second thought you may be able to do it separately but in that case you need to check the return of chmod.

Reply:
Yes that is what I thought, that I missed to check the return and throw an exception property if there was an error. I’ll check if it is better with mkdir.

Reply:
@pjanotti mkdir throws when a directory already exists, and Directory.CreateDirectory doesn't, it just ignores it and returns the existing directory. So I could do a check to see if the directory exists and use mkdir, or just leave it with chmod as I have it. Which one would you prefer and you think would be better.

Reply:
I am surprised that our Interop throws in case of error, does ChMod too? Anyway, it seems ok to keep the two steps separate as long as you check the results of the ChMod, or it throws in case of error. However, going this path is a bit tricky to confirm the behavior in some corner cases, for example: other user creates the folder and the domain socket with the same name, what if they open a handle to it prior to the ChMod of the folder? Is it detected later at the socket creation time? Probably it is ok but I don't know...

Typically doing a single operation simplifies understanding such scenarios. In this regard a simpler approach is to use the pipe name in the folder name and create it every time, letting MkDir throw if it already exists , e.g.: /tmp/.NET1000MyPipeName/p ... that said this has implications for cleanup and other parts of the code... sorry for thinking about it this late... perhaps best go the current route anyway...

Reply:
Since it is interop code it would not throw but would return an error code, that is what I meant with throw :p! So we would have to get the error and look if the error is that the directory already exists. If not, throw. But adding the pipe name to the directory would take space from the socket in Mac :s

Reply:
you said that it throws so I believe it 😛  but you don't need to use the name anywhere else in the path you would just switch from /tmp/.NET1000/MyNamedPipe to /tmp/.NET1000MyNamedPipe/p no?

Reply:
But do we really want to not name the pipe file as the pipe name and move it into the directory? So if we go with this route when disposing the server then we would need to remove the directory, right?

Reply:
Right, that's why I said up that there are other implications, and perhaps it is too big of a change to do right now... it need to be thought out